src/shared/services/validation.js

```diff
@@ -7,6 +7,117 @@ const containsForbiddenChars = (str, forbiddenChars) => {
     return false;
 };
 
+export const validateUsername = (username) => {
+    const invalidChars = [' ', ',', ';', '<', '>', '(', ')', '[', ']', '\\', '\"', '@', '$', '%', '^', '*'];
+    const minLength = 3;
+    const maxLength = 20;
+    const validation = {
+        isValid: true,
+        message: '',
+    };
```

Comment:
Use the `validateUsername` function to return the empty message if it's valid, or the error message otherwise.

Instead of 

```js
const validation = {
        isValid: true,
        message: '',
    };
```
use this

```js
let message = '';
```

Return either an empty message if the username is valid, or a string with an error message otherwise. 
Use this approach for the `validateEmail`, `validateUsernameOrEmail`, `validatePassword` functions as well.